src/renderer/hooks/store.js

```diff
@@ -86,11 +107,26 @@ export const useExperimentsFlagsStore = createPersistedStore(
   experimentsFlagsStoreSlice,
   'experiments-flags'
 )
-export const useBackgroundMapStore = createPersistedStore(
+
+const useBackgroundMapState = createPersistedStore(
   backgroundMapStoreSlice,
   'background-maps'
 )
 export const usePersistedUiStore = createPersistedStore(
   persistedUiStoreSlice,
   'ui'
 )
+
+export const useBackgroundMapStore = () => {
```

Comment:
Now this hooks encapsulates the logic for whether background maps is enable or not. And can only show and set the maps that are relevant to that state. No need to the developer to do a check everytime.